viewing_party/party.py

```diff
@@ -1,23 +1,213 @@
 # ------------- WAVE 1 --------------------
 
-def create_movie(title, genre, rating):
-    pass
+from symbol import single_input
+
+
+def create_movie(title, genre, rating): #Wave 1, step 1 done
+    movie_dict={}
+    if title is None or genre is None or rating is None:
+        return None
+
+    movie_dict["title"]=title
+    movie_dict["genre"]=genre
+    movie_dict["rating"]=rating
+    return movie_dict
+
+def add_to_watched(user_watched_dict,movie_info):
+    user_watched_dict["watched"] = [movie_info]
+    return user_watched_dict
+
+def add_to_watchlist(user_watchlist_dict, movie_info):
+    user_watchlist_dict["watchlist"] = [movie_info]
+    return user_watchlist_dict
+
+def watch_movie(single_user_data,movie_title):    
+    list_of_watchlist_movie_info=single_user_data["watchlist"]
+    list_of_watched_movie_info=single_user_data["watched"]
+
+    #Adds movie info to list of watched movies if it is in the watchlist and then removes the movie info from the watchlist
+    for index,movie_info in enumerate(list_of_watchlist_movie_info):
+        if movie_title == movie_info["title"]:
+            list_of_watched_movie_info.append(movie_info)
+            list_of_watchlist_movie_info.pop(index)
+    return single_user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(single_user_data):
+    list_of_watched_movie_info=single_user_data["watched"]
+    
+    #Calculates the number of watched movies
+    number_of_watched_movies=len(list_of_watched_movie_info)
+
+    list_of_movie_ratings=[]
+    sum_of_all_ratings=0
+
+    if len(list_of_watched_movie_info) == 0:
+        return 0.0
+
+    #Adds rating of each watched movie to a list
+    for movie_info in list_of_watched_movie_info:
+        list_of_movie_ratings.append(movie_info["rating"])
+
+    #Determines the sum of all ratings
+    for rating in list_of_movie_ratings:
+        sum_of_all_ratings+=rating

```

Comment:
How could we combine these two for loops? We are already looping over each movie on line 53, why not add its rating to `sum_of_all_ratings` right there instead of in a separate for loop?

This would also save us from creating `movie_ratings` altogether